tensorflow/core/kernels/cwise_op_add_1.cc

```diff
@@ -36,11 +36,7 @@ REGISTER_KERNEL_BUILDER(Name("Add")
 
 
 #if TENSORFLOW_USE_SYCL
-REGISTER(BinaryOp, SYCL, "Add", functor::add, float);
-
-#ifndef SYCL_NO_DOUBLE
-REGISTER(BinaryOp, SYCL, "Add", functor::add, double);
-#endif
+REGISTER_FLOATING(BinaryOp, SYCL, "Add", functor::add);
```

Comment:
Assuming that we added macros to (https://github.com/tensorflow/tensorflow/pull/11234/commits/f9845d72f605ae83b7511d7dd39d2c79221f5bc4#r126394733)
like  `TF_CALL_SYCL_NUMBER_TYPES` - that could be somehow composed - depending on the compiler flag - to include / exclude some types ( eg double or half )

we should be able to replace REGISTERx macro with something like:

```
#define REGISTER_KERNEL(type)                                  \
REGISTER(BinaryOp, SYCL, "Add", functor::add, type);
TF_CALL_SYCL_NUMBER_TYPES(REGISTER_KERNEL);
```
Suggested (https://github.com/tensorflow/tensorflow/pull/11234/commits/f9845d72f605ae83b7511d7dd39d2c79221f5bc4#diff-0c5a090f03366bec631c68d29f852ceeR490) will work for cwise ops but other ops (eg. conv, pooling ) would benefit from adding `TF_CALL_SYCL_NUMBER_TYPES` to `register_types.h`

What do you think?